subprojects/language-groovy/src/main/java/org/gradle/api/tasks/compile/GroovyCompile.java

```diff
@@ -276,8 +274,15 @@ private static void validateIncrementalCompilationOptions(List<File> sourceRoots
         }
     }
 
+    @Nullable
+    private JavaInstallationMetadata getToolchain() {
+        return javaLauncher.map(JavaLauncher::getMetadata).getOrNull();
+    }
+
     private GroovyJavaJointCompileSpec createSpec() {
-        DefaultGroovyJavaJointCompileSpec spec = new DefaultGroovyJavaJointCompileSpecFactory(compileOptions).create();
+        validateConfiguration();
+        DefaultGroovyJavaJointCompileSpec spec = new DefaultGroovyJavaJointCompileSpecFactory(compileOptions, getToolchain()).create();
+        assert spec != null;
```

Comment:
this can never happen

Reply:
Well the `Factory` contract says it can. So putting that here to get rid of null warnings in the IDE.

Reply:
Nevermind, I missed the `create`